### PR TITLE
Migrate to repo-config

### PR DESCRIPTION
**Pull request by llucax**
Some of the changes were made in separate commits, but at some point a lot of changes had to be grouped together because they are just too interdependant.

* Fix Python header
* Update license formatting
* Include hidden files in patterns by default
* Move security warning
* Update URL to avoid ambiguity
* Improve labeler configuration
* Remove unused labels from keylabeler
* Improve wording in release notes and template
* Fix typo in release notes and template
* Reformat the feature issue template
* Update bug issue template

* Add cookiecutter replay file to easy upgrading the repo-config templates.

* Migrate from `tox` to `nox`.

* Move development dependencies to optional dependencies in the `pyproject.toml` file.

* Add many other commmon configuration to the `pyproject.toml` file.

* Make formatting of markdown files consistent with other projects.

* Improve contributing guide with the dev dependencies improvements and to describe how to use `nox` and the docs generation.

* Add badges and improve the readme.

* Add mkdocs documentation pages and generation.

* Improve the CI to match the template, this includes:

  * Using a new action for protolint that will add inline comments in the PR diffs.
  * Add support for merge queues.
  * Use `nox`.
  * Use the dev dependencies defined in the `pyproject.toml` file.
  * Add and improve names of steps and jobs.
  * Add jobs for testing and publishing the generated documentation.
  * Remove the creation of a discussion for each release.

* Use `from x import y` instead of `import x.y as y`
* Disable import outside toplevel pylint check
* Add missing comments to test functions

### CODE REVIEW

**Comment by llucax** [comment]
OK, this worked locally but it is failing in the CI, I need to see what's happening but tomorrow :)

**Comment by llucax** [comment]
I think the bump of the dependency should fix it.

**Comment by llucax** [comment]
Enabling auto-merge.

File: .cookiecutter-replay.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      "local_extensions.introduction",
+      "local_extensions.keywords",
+      "local_extensions.pypi_package_name",
+      "local_extensions.python_package",
+      "local_extensions.src_path",
+      "local_extensions.title"
+    ],
+    "_template": "gh:frequenz-floss/frequenz-repo-config-python"
+  }
+}
```
Comment (tiyash-basu-frequenz) [inline .cookiecutter-replay.json:31]: Missing newline

**Review by tiyash-basu-frequenz — COMMENTED** [review]
A minor comment, but LGTM overall.

File: .cookiecutter-replay.json (hunk 1 above)
Comment (llucax) [reply .cookiecutter-replay.json:31]: This file is auto-generated, so I prefer to leave it as it is.

**Review by tiyash-basu-frequenz — APPROVED** [review]

**Comment by llucax** [comment]
OK, tests are passing now. I will need to update the branch protection rules to use the new check names, other PRs will need to be rebased.

**Comment by llucax** [comment]
Updated rules and enabled the merge queue for `v0.x.x`.

**Pull request merged**